Sources/Diagnostics/DiagnosticsTracker.swift

```diff
@@ -31,6 +31,13 @@ protocol DiagnosticsTrackerType {
                                    resultOrigin: HTTPResponseOrigin?,
                                    verificationResult: VerificationResult) async
 
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func trackPurchaseRequest(wasSuccessful: Bool,
+                              storeKitVersion: StoreKitVersion,
+                              errorMessage: String?,
+                              errorCode: Int?,
+                              storeKitErrorDescription: String?) async
```

Comment:
Should we also track the request time? It could be useful to have a graph of how long these requests take.

Reply:
The issue is that the request time would include the time the purchase dialog is displayed, making it useless. I don't think we have any other way to track this? We could track post receipts only?

Reply:
Yeah that's true ignore me. We should already been tracking the post receipt calls in the `http_request_performed` event so not needed!